src/ATen/native/sparse/xpu/SparseTensorMath.cpp

```diff
@@ -26,4 +35,68 @@ Tensor _sparse_sum_backward_xpu(
   return xpu::_sparse_sum_backward_kernel(grad_, input_, dims_to_sum);
 }
 
+Tensor& s_addmm_out_sparse_dense_xpu(Tensor& r_, const Tensor& t, const SparseTensor& sparse_, const Tensor& dense, const Scalar& beta, const Scalar& alpha) {
+  TORCH_CHECK(t.is_xpu(), "Expected all tensors to be on the same device. addmm: expected 'self' to be XPU, but got CPU");
+  TORCH_CHECK(r_.is_xpu(), "Expected all tensors to be on the same device. addmm: expected 'out' to be XPU, but got CPU");
+  TORCH_CHECK(sparse_.is_xpu(), "Expected all tensors to be on the same device. addmm: expected 'mat1' to be XPU, but got CPU");
+  TORCH_CHECK(dense.is_xpu(), "Expected all tensors to be on the same device. addmm: expected 'mat2' to be XPU, but got CPU");
+
+  // TORCH_CHECK(xpu::check_device({sparse_, r_, t, dense}));
```

Comment:
`xpu::check_device` can be found in `sparse/xpu/sycl/SparseTensorMathKernels.cpp`.